e2e/test/iothub/service/MessageFeedbackReceiverE2ETest.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+using System.Text;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.Azure.Devices.Client;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Azure.Devices.E2ETests.iothub.service
+{
+    /// <summary>
+    /// E2E test class for MessageFeedbackReceiver.
+    /// </summary>
+    [TestClass]
+    [TestCategory("E2E")]
+    [TestCategory("IoTHub")]
+    public class MessageFeedbackReceiverE2eTest : E2EMsTestBase
+    {
+        private bool messagedFeedbackReceived;
+        [TestMethod]
+        public async Task MessageFeedbackReceiver_Operation()
+        {
+            using var serviceClient = new IotHubServiceClient(TestConfiguration.IoTHub.ConnectionString);
+            serviceClient.MessageFeedbackProcessor.MessageFeedbackProcessor = feedbackCallback;
+            await serviceClient.MessageFeedbackProcessor.OpenAsync().ConfigureAwait(false);
+            var message = new Message(Encoding.UTF8.GetBytes("some payload"));
+            message.Ack = DeliveryAcknowledgement.Full;
+            messagedFeedbackReceived = false;
+            await serviceClient.Messaging.SendAsync(TestConfiguration.IoTHub.X509ChainDeviceName, message).ConfigureAwait(false);
+            await receiveMessage().ConfigureAwait(false);
+
+            var timer = Stopwatch.StartNew();
+            while (timer.ElapsedMilliseconds < 60000)
```

Comment:
Don't just wait until a minute has passed. You should check to see if the feedback message has been received here as well. If the feedback message has been received, then you can exit this loop since the test passed

Reply:
It also helps test failures be more descriptive if you explicitly throw an assertion failure saying "Timed out waiting to receive message feedback" once this loop completes and the message feedback wasn't received